File: dotnet/Multiplayer/Players/PlayerRenameResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Players
{
    /// <summary>
    /// Response returned when a player is successfully renamed.
    /// </summary>
    public class PlayerRenameResponse : ApiResponse<EPlayerRenameError>
    {
        /// <summary>
        /// Confirmation message for the name change.
        /// </summary>
        public string Message { get; set; } = string.Empty;

        /// <summary>
        /// The new name assigned to the player.
        /// </summary>
        public string New_name { get; set; } = string.Empty;

        /// <summary>
        /// The ID of the player who was renamed.
        /// </summary>
        public int Player_id { get; set; }
    }
}

File: java/src/main/java/com/michitai/multiplayer/players/PlayerRenameResponse.java
package com.michitai.multiplayer.players;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response confirming a player name change.
 */
public class PlayerRenameResponse extends ApiResponse {
    @JsonProperty("message")
    private String message;

    @JsonProperty("new_name")
    private String newName;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getNewName() {
        return newName;
    }

    public void setNewName(String newName) {
        this.newName = newName;
    }
}
